openedx/core/lib/block_structure/block_structure.py

```diff
@@ -66,11 +66,11 @@ def __init__(self, root_block_usage_key):
 
     def __iter__(self):
         """
-        The default iterator for a block structure is a topological
-        traversal since it's the more common case and we currently
-        need to support DAGs.
+        The default iterator for a block structure is get_block_keys()
+        since we need to filter blocks as a list.
+        A topological traversal can be used to support DAGs.
```

Comment:
@nasthagiri Can you help clarify when you get the chance? Is a block_structure a block_dag?  My understanding was that the results of this process will no longer be a block_structure, but the transformers still need it.  Are we introducing a new block_list or something like that?  I'm afraid I'm not being too helpful here without you.

Thanks again @lenacom.  Hopefully @nasthagiri can help clarify soon.